### ISSUE TITLE
3.0.9 socket takes longer to subscribe to multiple markets than 2.1.20

### ISSUE THREAD

**Issue by ghost** [JKorf/Bittrex.Net#147]
```c#
            var socket = new BittrexSocketClient();
            var markets = new BittrexClient().GetMarkets().Data;
            foreach (var market in markets)
            {
                var result = await socket.SubscribeToExchangeStateUpdatesAsync(market.MarketName, onUpdate =>
                {
                    onUpdate.Fills.ForEach(fill =>
                    {
                        WriteLine($"{market.MarketName} {fill.Rate}");
                    });
                });
                WriteLine($"{market.MarketName} Subscribed {result.Success }");
            }
```
Bittrex.Net 3.0.9 socket takes longer (perhaps 10x) to subscribe to multiple markets than Bittrex.Net 2.1.20
I'm using dotnetcore 2.2. 

edit: 3.0.1 has the same speed issue as 3.0.9. I'm not sure about 3.0.0 as that version doesn't work at all for me.

**Comment by JKorf** [comment]
Hi, you're right. This is because in newer versions a new socket connection is created for every subscription. However, Bittrex uses SignalR which interlocks connection calls and takes at least a second for each of them. Meaning, if you have a lot of subscriptions to make at once it will be slow. I'm looking into combining sockets for multiple subscriptions, but don't have a time table on that. 
When API version v3 will be available this will be fixed as well since Bittrex won't be using SignalR for that.

**Comment by pwnzya** [comment]
Speaking of API v3, they just opened it up for everyone (beta release), although the websock part isn't added yet in the document... [Bittrex Changelist 04/19/2019](https://bittrex.zendesk.com/hc/en-us/articles/360027160871-Changelist-04-19-2019)
Hopefully, it'll come soon.

**Comment by JKorf** [comment]
Hi, I've implemented support for the V3 version in the latest 3.1.1 release, it's accessable via the `BittrexClientV3` client. 

On topic; I've reworked the socket backend in the 3.1.0 release, this should mean that subscribing to multiple markets should be considerably faster. In the new version multiple subscriptions (default 10) use a single socket connection meaning that it doesn't have to connect a new socket for every subscription.
